x/auth/ante/fee.go

```diff
@@ -25,68 +27,107 @@ type DeductFeeDecorator struct {
 	bankKeeper     types.BankKeeper
 	feegrantKeeper FeegrantKeeper
 	txFeeChecker   TxFeeChecker
+
+	// pointer to a separate state struct
+	state *deductFeeState
 }
 
-func NewDeductFeeDecorator(ak AccountKeeper, bk types.BankKeeper, fk FeegrantKeeper, tfc TxFeeChecker) DeductFeeDecorator {
-	if tfc == nil {
-		tfc = checkTxFeeWithValidatorMinGasPrices
-	}
+// deductFeeState holds the mutable state needed across method calls
+type deductFeeState struct {
+	minGasPrices   sdk.DecCoins
```

Comment:
can we pull this into the main struct as it wont change between txs?

Reply:
I tried. But since we're setting txFeeChecker during initialization, the MinGasPrices value in the associated decorator struct isn't getting updated. To solve this, I created a separate struct and used a pointer to connect it to the decorator struct.